src/main/java/com/mparticle/kits/TaplyticsKit.java

```diff
@@ -0,0 +1,300 @@
+        options.put("delayedStartTaplytics", true);
+
+        if (options != null) {
+            Taplytics.startTaplytics(context, apiKey, options);
+            return;
+        }
+
+        Taplytics.startTaplytics(context, apiKey);
+    }
+
+    private String getAPIKey(Map<String, String> settings) {
+        final String apiKey = getSettings().get(API_KEY);
+        if (TextUtils.isEmpty(apiKey)) {
+            throw new IllegalArgumentException(API_KEY);
+        }
+        return apiKey;
+    }
+
+    /**
+     * Get Taplytics options from settings
+     * @param settings
+     * @return
+     */
+
+    private Map<String, Object> getOptionsFromConfiguration(Map<String, String> settings) {
+        Map<String, Object> options = new HashMap<>();
+        addAggressiveOption(options, settings);
+
+        return options.isEmpty() ? null : options;
+    }
+
+    private void addAggressiveOption(Map<String, Object> options, Map<String, String> settings) {
+        Boolean agg = Boolean.parseBoolean(settings.get(AGGRESSIVE));
+        options.put("aggressive", agg.booleanValue());
+    }
+
+    @Override
+    public String getName() {
+        return "Taplytics";
+    }
+
+    /**
+     * AttributeListener Interface
+     */
+
+    @Override
+    public void setUserAttribute(String attributeKey, String attributeValue) {
+        try {
+            JSONObject attr = new JSONObject();
+            attr.put(attributeKey, attributeValue);
+            Taplytics.setUserAttributes(attr);
+        } catch (JSONException e) {
+
+        }
+    }
+
+    @Override
+    public boolean supportsAttributeLists() { return false; }
+
+    @Override
+    public void setAllUserAttributes(Map<String, String> attributes, Map<String, List<String>> attributeLists) {
+        for (Map.Entry<String, String> entry : attributes.entrySet()) {
+            setUserAttribute(entry.getKey(), entry.getValue());
+        }
+    }
+
+    @Override
+    public void setUserIdentity(MParticle.IdentityType identityType, String identity) {
+        switch (identityType) {
+            case CustomerId: {
+                setUserAttribute(USER_ID, identity);
+                break;
+            }
+            case Email: {
+                setUserAttribute(EMAIL, identity);
+            }
+        }
+    }
+
+    @Override
+    public void removeUserIdentity(MParticle.IdentityType identityType) {
+        setUserIdentity(identityType, null);
+    }
+
+    /*
+        Unsupported methods
+     */
+    @Override
+    public List<ReportingMessage> logout() { return null; }
+
+    @Override
+    public void removeUserAttribute(String attribute) { }
+
+    @Override
+    public void setUserAttributeList(String attribute, List<String> attributeValueList) { }
+
+    /**
+     * CommerceListener Interface
+     */
+
+    @Override
+    public List<ReportingMessage> logEvent(CommerceEvent event) {
+        TransactionAttributes transactionAttributes = event.getTransactionAttributes();
```

Comment:
you should check if this is a Purchase event here vs something else like an add to cart or impression, etc.

See here for an advanced example: https://github.com/mparticle-integrations/mparticle-android-integration-appboy/blob/master/src/main/java/com/mparticle/kits/AppboyKit.java#L151